File: ToDoApplication/Models/ToDoViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ToDoApplication.Models
{
    public class ToDoViewModel
    {
        public ToDoViewModel()
        {
            Id = 0;
        }
        public int Id { get; set; }

        [Required(ErrorMessage = "Title is Required")]
        [MaxLength(100)]
        public string Title { get; set; }

        [MaxLength(1000)]
        public string Description { get; set; }
        
        [Range(1,5)]
        [Required]
        public int? Priority { get; set; }

        [EmailAddress]
        public string Email { get; set; }

        public DateTime  CreatedOn  { get; set; }
    }
}

File: ToDoApplication/Scripts/ToDoViewModel.js
function ToDoViewModel(title, description, email, priority) {
    var self = this;
    self.title = title || "";
    self.description = description || "";
    self.email = email || "";
    self.priority = priority || "";
}

    /*
     1.Title Required, StringLength (100) 
     2. Description StringLength (1000)
     3. Mail To Presence, EmailsOnly 
     4. Priority Range(1 - 5)
    */